test/lib/jdk/test/lib/process/ProcessTools.java

```diff
@@ -174,7 +177,16 @@ synchronized int readNext() {
             }
             while (current == count) {
                 if (!p.isAlive()) {
-                    return -1;
+                    try {
+                        // It is needed to wait until stream is flushed after
+                        // process is completed.
+                        task.get();
```

Comment:
This looks problematic - if you block here you are holding the object monitor as this is a synchronized method.

Reply:
You are right. It is needed to give a chance to write to this buffer. fixed.